### PR TITLE
Scale miner output with power

### PR DESCRIPTION
**Pull request by adacovsk**
See #33135. Had an issue with the bot. Some sort of map conflict. Redoing PR.

## Changelog

:cl:
 * rscadd: Gas miner output scales with power output

### CODE REVIEW

**Comment by pedr69** [comment]
one problem i see is that atmos(on most maps) is on the main station powernet, not the engineering one
having it suck all excess power always is pretty stupid in my opinion

**Comment by adacovsk** [comment]
> one problem i see is that atmos(on most maps) is on the main station powernet, not the engineering one having it suck all excess power always is pretty stupid in my opinion

Ahh you're right. I actually meant to keep it at using around 50% the surplus rather than max it out

**Comment by adacovsk** [comment]
> one problem i see is that atmos(on most maps) is on the main station powernet, not the engineering one having it suck all excess power always is pretty stupid in my opinion

It's not as big of an issue as you might think, since its power priority is the lowest

**Comment by DorainGrey** [comment]
The power pulled should be adjustable via multitool or some other method, I shouldn't need to pump twice the power required into the grid, to produce more gasses because it only takes 50% of excess.

**Comment by Kurfursten** [comment]
Dorain is fully right on this one. Needs to be configurable how much juice you want to pump into it.

**Comment by adacovsk** [comment]
> Dorain is fully right on this one. Needs to be configurable how much juice you want to pump into it.

It uses jellyveggie2's power priority, so it will only take excess in the connected powernet. This is the best I can do right now

**Comment by Kurfursten** [comment]
I can show you how to make a multitool menu

**Comment by adacovsk** [comment]
> I can show you how to make a multitool menu

yeah okay

**Comment by PrimeDSS13** [comment]
Won't this make pressurizing a new location such as the derelict or a custom built locale much much harder due to the lack of power?

**Comment by adacovsk** [comment]
> Won't this make pressurizing a new location such as the derelict or a custom built locale much much harder due to the lack of power?

They already require power to be on, so no. I scaled the pressure output to be on par at about 150 kW. So any engine will output the expected amount of pressure, other than PACMAN. Even as such, an output of 100kPa from the miner is enough to pressurize anything reasonably over time.

**Comment by DorainGrey** [comment]
> 
So there's a baseload generation that doesn't suck any excess power? 
What's the output pressure at room temp from 30kw input for example? That's about the excess on box with an AME.

**Comment by adacovsk** [comment]
> So there's a baseload generation that doesn't suck any excess power? What's the output pressure at room temp from 30kw input for example? That's about the excess on box with an AME.

There is no baseload because I thought having consequences of no power seems fun. 30kW is 3000kPa. Currently, the miners without this PR just output 4500kPa. With a singulo you're probably generating even more already

**Comment by pedr69** [comment]
https://github.com/vgstation-coders/vgstation13/blob/9a306ca45aa55eb51e7dc87798d90bc2ab979a46/code/game/machinery/atmoalter/gas_mine.dm#L57
oh, and why not add the power draw here, so you can see how fast they're pumping when you examine em

**Comment by adacovsk** [comment]
> > miners will now need 150kw each to be as fast they were before
> 
> now you need 5*150kw=750KW to cool plasma at the old rate, since you can't prioritise the plasma miner i'd add a way to change their power priority
> 
> this also removes the 10MPA max pressure limit on the miners?
>oh, and why not add the power draw here, so you can see how fast they're pumping when you examine em

ahh shit you're right, I calculated wrong.. I'll scale it differently. Very good ideas. This should also remove the max pressure limit, yes.

And you're right, there should be a way to change the power priority, that is another great idea. I'll look into it.

**Comment by pedr69** [comment]
>Ceiling(0.5 * power_load)

magic numbers detected
#define GASMINER_JOULES_PER_CREATED_KILOPASCAL?
could also have a "rate" var exist always, and be set in update_rate(), so you can use that when displaying, instead of only having it defined inside update_rate()

man, it's hard to explain what i mean, is it because i'm an ESL and haven't read programming books, or is it like that for everyone?

**Comment by Kurfursten** [comment]
No, I think your statement is reasonable. You're asking for the same thing that Dorain and I are - the ability to configure how much power draw these take instead of having it just always be half of power_load

File: code/game/machinery/atmoalter/gas_mine.dm [hunk 1]
```diff
@@ -41,14 +74,15 @@
 	if(stat & BROKEN)
 		to_chat(user, "<span class='info'>\The [src]'s status terminal reads: Broken.</span>")
 		return
-	to_chat(user, "<span class='info'>\The [src]'s status terminal reads: Functional and operating.</span>")
+	to_chat(user, "<span class='info'>\The [src]'s status terminal reads: Functional and operating at a rate of [Ceiling(0.5 * power_load)] kPa per cycle.</span>")
```
Comment (pedr69) [inline code/game/machinery/atmoalter/gas_mine.dm:77]: magic number detected

**Comment by pedr69** [comment]
https://github.com/vgstation-coders/vgstation13/blob/d6363976bfe887800ae402e858b55443cadac1d6/code/game/machinery/atmoalter/gas_mine.dm#L77
magic numbers...

File: code/game/machinery/atmoalter/gas_mine.dm [hunk 2]
```diff
@@ -1,3 +1,5 @@
+#define KW_TO_MOLS_COEFFICIENT 0.5
```
Comment (pedr69) [inline code/game/machinery/atmoalter/gas_mine.dm:1]: the units are wrong
joules to moles, or watts to moles per second

kind of a moot point, half our power units are fucked

Comment (pedr69) [reply code/game/machinery/atmoalter/gas_mine.dm]: >makes the name even more retarded

based?

**Comment by pedr69** [comment]
imagine the ZAS stuns...
even now, with 10MPA pressure checks, an atmos tech entering the plasmaminer chamber without maggies on gets stunned for a good 2-3 minutes
imagine someone doing that at 14:00, when the pressure inside is several gigapascals

**Comment by adacovsk** [comment]
> imagine the ZAS stuns... even now, with 10MPA pressure checks, an atmos tech entering the plasmaminer chamber without maggies on gets stunned for a good 2-3 minutes imagine someone doing that at 14:00, when the pressure inside is several gigapascals

I put a cap on the ZAS stuns, so no worries

**Comment by jwhitak** [comment]
So uh what happens when they're wrenched in space? The place where power isn't required? (Miners work in space.)

**Comment by adacovsk** [comment]
> So uh what happens when they're wrenched in space? The place where power isn't required? (Miners work in space.)

they need a powernet, so the rate will be 0 unless there's a powered APC... Could change it so that if there's no powernet it has a base rate

**Comment by pedr69** [comment]
any day now....

File: code/game/machinery/atmoalter/gas_mine.dm [hunk 3]
```diff
@@ -111,138 +158,64 @@
...
+	gases = list(GAS_CARBON = 1)
 
 /obj/machinery/atmospherics/miner/air
 	name = "\improper Air Miner"
 	desc = "You fucking <em>cheater</em>."
 	overlay_color = "#70DBDB"
-
-	on = 0
-
-/obj/machinery/atmospherics/miner/air/AddAir()
```
Comment (ShiftyRail) [inline code/game/machinery/atmoalter/gas_mine.dm:191]: Why are all these deleted?

**Comment by ShiftyRail** [comment]
This just looks like pure griefing to me but /shrug. Let's see how it works.

**Pull request merged**